compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -428,6 +428,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let ptr_ty = self.resolve_vars_if_possible(ptr_ty);
                         variadic_error(tcx.sess, arg.span, arg_ty, &ptr_ty.to_string());
                     }
+                    // generics? in C variadic? naaah
+                    ty::Adt(_def, gen_args) if gen_args.len() > 0 => {
+                        bug!("what the ferris are you doing")
+                    }
+                    ty::Param(_) => bug!("what the ferris are you doing"),
```

Comment:
Why are we ok w/ `[T; N]` 🤔

Reply:
Let me be clear here that I mean this for example. This also applies to all the other built-in types that allow other types inside of them, like tuples, refs, slices, etc.

Reply:
we probably aren't!